### ISSUE TITLE
SpikeMonitor should support recording sets of indices

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2#1236]
See https://brian.discourse.group/t/how-to-record-only-certain-neurons-from-a-group-by-spikemonitor/167/5

I think we avoided doing this in the past, but if we are going to implement https://github.com/brian-team/brian2/pull/1229 then why not this too?

**Comment by mstimberg** [comment]
Agreed, if we do #1229, then this would just be some minor refactoring so that record indices and non-contiguous subgroups are handled in the same way. Which reminds me that there is a subtle issue about what indices to use in that case, but let's discuss this in #1229 .
